fetch.bs

```diff
@@ -3342,86 +3345,149 @@ steps:
-   <a>network error</a>.
+    <ol>
+     <li><p>Update <var>storedResponse</var>'s <a for=response>header list</a> using
+     <var>forwardResponse</var>'s <a for=response>header list</a>, as per the 
+     "<a href=https://tools.ietf.org/html/rfc7234#section-4.3.4>Freshening Stored Responses upon Validation</a>"
+     chapter of <cite>HTTP Caching</cite> [[!HTTP-CACHING]].
 
-   <li><p>Update <var>cachedResponse</var>'s <a for=response>header list</a>
-   using <var>response</var>'s <a for=response>header list</a>, as per the
-   "<a href=https://tools.ietf.org/html/rfc7234#section-4.3.4>Freshening Stored Responses upon Validation</a>"
-   chapter of <cite>HTTP Caching</cite>. [[!HTTP-CACHING]]
+     <li><p>Set <var>response</var> to <var>storedResponse</var>.
+
+     <p class="note">This updates the stored response in cache as well.
+    </ol>
 
    <li>
-    <p>Set <var>response</var> to the <var>cachedResponse</var>.
+    <p>If the <var>partial</var> flag is set and <var>forwardResponse</var>'s 
+    <a for=response>status</a> is <code>206</code>, run these substeps:
 
-    <p class="note no-backref">This changes <var>response</var> entirely, including its
-    <a for=response>status</a> which is most likely <code>200</code> now.
+    <ol>
+     <li><p>Update <var>storedResponse</var>'s <a for=response>header list</a> using
+     <var>forwardResponse</var>'s <a for=response>header list</a>, as per the 
+     "<a href=https://tools.ietf.org/html/rfc7234#section-3.3>Combining Partial Content</a>" chapter
+     of <cite>HTTP Caching</cite> [[!HTTP-CACHING]].
+
+     <li><p>Set <var>response</var> to <var>storedResponse</var>.
+
+     <p class="note">This updates the stored response in cache as well.
+    </ol>
+    <!-- TODO: account for partial request failure -->
+
+    <li><p>If <var>response</var> is null, run these substeps:
+
+     <ol>
+      <li><p>If <var>forwardResponse</var> is a <a>network error</a>, <var>storedResponse</var> is
+      not null, <var>storedResponse</var> is complete, and <var>storedResponse</var> is allowed to 
+      be served stale, as per the
+      "<a href=https://tools.ietf.org/html/rfc7234#section-4.2.4>Serving Stale Responses</a>"
+      chapter of <cite>HTTP Caching</cite> [[!HTTP-CACHING]]>", set <var>response</var> to
+      <var>storedResponse</var> and abort these substeps.
+
+      <li><p>Set <var>response</var> to <var>forwardResponse</var>.
+
+      <li><p>Store <var>httpRequest</var> and <var>forwardResponse</var> in the HTTP cache, as per 
+      the "<a href=https://tools.ietf.org/html/rfc7234#section-3>Storing Responses in Caches</a>"
+      chapter of <cite>HTTP Caching</cite> [[!HTTP-CACHING]].
+
+      <p class=note>The cache key used to store the response must include the value of the
+      <var>credentials</var> flag, to ensure that a stored response to a request with credentials is
+      not used to satisfy a request without credentials.
```

Comment:
Same comment as earlier, no requirements in notes. Also a formatting glitch here due to multiple `<p>` inside the `<li>`.